### ISSUE TITLE
Feature request: Jump to definition, Ctrl-Click.

### ISSUE THREAD

**Issue by Kiiyya** [meraymond2/idris-vscode#76]
It would be super handy to be able to ctrl-click go-to-definition on any identifier, if possible also on operators (since notation in idris2 is very frightening for beginners like me). The hover already works and is useful, but often I am interested in what else is contained in a given module. At least for me, go-to-definition is my Nr 1 feature in any language IDE extension. I'm using idris2.

Thanks for making this extension :)

**Comment by michaelmesser** [comment]
Go to definition is supported by [idris2-lsp-vscode](https://github.com/bamboo/idris2-lsp-vscode). However, installing my the LSP can be a bit tricky.

**Comment by meraymond2** [comment]
Hi, thanks for making an issue.

I'm not aware of any simple way to implement this. This extension is mostly a wrapper around [Idris’s IDE protocol](https://idris2.readthedocs.io/en/latest/implementation/ide-protocol.html), and I don't think it exposes the information I would need. I'll have a think though.

In the meantime, you could try the lsp extension michaelmesser recommended. The language server has some functionality that the IDE protocol doesn't, although the last time I looked it still had some missing features. I'm not sure how easy it is to run the two extensions side-by-side — I need to investigate that and update the readme.

**Comment by michaelmesser** [comment]
I don't think it would be a good idea to run the extensions side by side. I will have some time over the next month to work on improving the LSP so let me know what is missing.

**Comment by doofin** [comment]
@meraymond2 Would it be a good idea to implement this in idris2 compiler and add a new command in language protocols?
